easybuild/framework/easyblock.py

```diff
@@ -3350,6 +3352,271 @@ def _sanity_check_step_multi_deps(self, *args, **kwargs):
+                                    self.log.warning(fail_msg, highest_cc[0], path, derived_ptx_ccs)
+                                elif accept_missing_ptx:
+                                    self.log.warning(fail_msg, highest_cc[0], path, derived_ptx_ccs)
+                                else:
+                                    files_missing_ptx_fails.append(path)
+                                    num_files_missing_ptx_fails += 1
+                                    fails.append(fail_msg % (highest_cc[0], path, derived_ptx_ccs))
+                            else:
+                                msg = (f"Output of 'cuobjdump' checked for '{path}'; ptx code was present for (at "
+                                       "least) the highest CUDA compute capability in cuda_compute_capabilities")
+                                self.log.debug(msg)
+                else:
+                    self.log.debug(f"Not sanity checking files in non-existing directory {dirpath}")
+
+            # Summary
+            summary_msg = "CUDA sanity check summary report:\n"
+            summary_msg += f"Number of CUDA files checked: {num_cuda_files}\n"
+            summary_msg += f"Number of files missing one or more CUDA Compute Capabilities: {num_files_missing_cc} "
+            summary_msg += f"(ignored: {num_files_missing_cc_ignored}, fails: {num_files_missing_cc_fails})\n"
+            if accept_ptx_as_cc:
+                summary_msg += "Number of files missing one or more CUDA Compute Capabilities, but has suitable "
+                summary_msg += "PTX code that can be JIT compiled for the requested CUDA Compute Capabilities: "
+                summary_msg += f"{num_files_missing_cc_but_has_ptx}\n"
+            summary_msg += f"Number of files with device code for more CUDA Compute Capabilities than requested: "
+            if strict_cc_check:
+                summary_msg += f"{num_files_surplus_cc} (ignored: {num_files_surplus_cc_ignored}, fails: "
+                summary_msg += f"{num_files_surplus_cc_fails})\n"
+            else:
+                summary_msg += f"{num_files_surplus_cc} (not running with --strict-cuda-sanity-check, so not "
+                summary_msg += "considered failures)\n"
+            summary_msg += "Number of files missing PTX code for the highest configured CUDA Compute Capability: "
+            if accept_missing_ptx:
+                summary_msg += f"{num_files_missing_ptx} (running with --accept-missing-cuda-ptx so not considered "
+                summary_msg += "failures)\n"
+            else:
+                summary_msg += f"{num_files_missing_ptx} (ignored: {num_files_missing_ptx_ignored}, fails: "
+                summary_msg += f"{num_files_missing_ptx_fails})\n"
+            if not build_option('debug'):
+                summary_msg += "Rerun with --debug to see a detailed list of files.\n"
+            # Give some advice
+            if num_files_missing_cc > 0 and not accept_ptx_as_cc:
```

Comment:
For tis and others below we should consider what to say when failures are being ignored by default (which is likely to be the default for EasyBuild).